src/jira/client/axios.ts

```diff
@@ -154,7 +154,19 @@ const instrumentFailedRequest = (baseURL: string, logger: Logger) => {
 			logger.info("Ok, looks like still works.");
 		}
 		instrumentRequest(error?.response);
+
 		if (error.response?.status === 503 || error.response?.status === 405) {
+			/**
```

Comment:
And, if possible, it should be done in the error handling part, not the `instrucmentXXX` part, coz by name instructment If posshould just send metrics instead of update the error (although existing code is also wrong)

```
	instance.interceptors.response.use(
		instrumentRequest,
		instrumentFailedRequest(baseURL, logger)
	);

	instance.interceptors.response.use(
		getSuccessMiddleware(logger),
		getErrorMiddleware(logger)
	);

```

Reply:
Moved the logic inside `getErrorMiddleware`.